app/services/match_service.rb

```diff
@@ -6,11 +6,18 @@ class MatchService
   # @param teams [Array] the teams to generate matches with
   # @return [Array] the generated matches
   def self.generate_matches(teams)
-    if teams.size < 2
+    if teams.empty?
       # should be prevented by controller
```

Comment:
I think throwing an exception **or** returning an empty Array is probably the better way to handle this.

Reply:
which one of the two?

Reply:
Throwing an exception is better imo.